src/OrchardCore.Modules/OrchardCore.ContentFields/Controllers/ContentPickerAdminController.cs

```diff
@@ -53,11 +54,24 @@ public async Task<IActionResult> SearchContentItems(string part, string field, s
                 return new ObjectResult(new List<ContentPickerResult>());
             }
 
+            IEnumerable<string> contentTypes = fieldSettings.DisplayedContentTypes;
```

Comment:
because `fieldSettings.DisplayedContentTypes` is an array. And by defining it as `IEnumerable<string>`, I avoided the need to call `.ToArray()` on the following LINQ statement.

Reply:
Why you need to avoid `ToArray()` while its already an array, you might need to add  using for `System.Linq`

Reply:
The `LINQ` query is IQueryable not array. Instead of explicitly use ToArray to execute the query and convert it to array, we just pass the query to the next call. I think it’s unnecessary operation to call array in this case.

Reply:
Yes, when possible I prefer to use var.

Hmm, but maybe better to use `.ToArray()`, not a fan to let Linq expressions not resolved, unless when used just after.

Didn't look at it in details but in the below Linq expression are you sure we need to use `StringComparer.OrdinalIgnoreCase`.

Reply:
Not sure if the ignore case is needed. In widget driver we use [==](https://github.com/OrchardCMS/OrchardCore/blob/9d8e75f5caaa0540e6206228f098980350dad244/src/OrchardCore.Modules/OrchardCore.Flows/Drivers/FlowPartDisplayDriver.cs#L142) so I’ll change that

Reply:
Updated.